Fix the media stream not being stopped; mirror the QR camera

katspaugh opened the pull request:
Fixes #316 and #2138

### What's been done
* Correctly stop the media stream that we create to check for camera permissions
* Use the front-facing mode so that the video is flipped horizontally
* Fix eslint warnings in the QR reader component

### How to test
Follow the steps in #316.
The camera will still occasionally linger for a couple seconds but I fixed that it sometimes didn't turn off at all.

katspaugh commented:
I have read the CLA Document and I hereby sign the CLA

fernandomg commented on src/components/ScanQRModal/utils.ts:
```diff
@@ -6,13 +6,17 @@ navigatorCp.getMedia =
-  )
+export const checkWebcam = (): Promise<boolean> => {
+  return new Promise((resolve) => {
+    navigatorCp.getMedia(
+      { video: true },
+      (stream: MediaStream) => {
+        stream.getTracks().forEach((track) => track.stop())
+        resolve(true)
+      },
+      () => {
+        resolve(false)
+      },
+    )
```
shouldn't we change this in favor of [`navigator.mediaDevices.getUserMedia`](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia) instead of the deprecated [`navigator.getUserMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/getUserMedia)?

It also has a promise-based interface instead of callbacks.

Further information: https://caniuse.com/?search=getUserMedia

katspaugh replied:
Good point, will do! 👍

fernandomg reviewed (changes requested):
@katspaugh, hmm it's getting unresponsive now

Modal blocked (chrome and firefox)
1. be sure to not have the camera allowed
2. trigger the qr-reader
3. don't allow the camera
4. the "upload file" dialog will popup, close the dialog
5. now you can't close the modal in any way
<br>

Multiple camera allowance requests (firefox)
1. be sure to not have the camera allowed
2. trigger the qr-reader
3. allow the camera
4. a new "will you allow" modal will appear
<br>

Multiple camera allowance requests and modal blocked (firefox)
1. do the same steps as the previous case
2. for the second "will you allow" modal, opt for "don't allow"
3. now you can't close the modal in any way
<br>

Due to the Firefox issues, I'm having second thoughts about the new API, WDYT?

nicosampler commented on src/components/ScanQRModal/index.tsx:
```diff
@@ -88,16 +91,17 @@ export const ScanQRModal = ({ isOpen, onClose, onScan }: Props): React.ReactElem
             legacyMode={!useWebcam}
-            onError={(err) => onFileScannedResolve(err, null)}
-            onScan={(data) => onFileScannedResolve(null, data)}
+            onError={(err: Error) => onFileScannedResolve(err, null)}
```
can we just pass the function here? `onError={onFileScannedResolve}`

katspaugh replied:
TypeScript will complain because the callback needs 2 arguments and `onError` only passes 1.

katspaugh commented:
@fernandomg thanks, I didn't check in Firefox.

I've removed the `checkWebcam` and just use the permission error from the QR code reader itself. Also fixed the modal not closing.

fernandomg reviewed (approved):
:1st_place_medal: nice!

francovenica commented:
Looks good to me.

I think the feature is the same for every input it is implemented. Still, I tested in
Safe loading
Send funds
Contract interaction
Address book
Adding owners
Spending limit beneficiary

Made sure the cam turns off in a second or 2 after:
Successfully recognizing the QR code
Opening the folder to pick a photo (Just by opening it. if you close the folder the cam doesn't turn on again, but that's a normal behavior)
When closing the cam modal (with the X or clicking outside of it)
